### DISCUSSION TITLE
feature: serve asset with file or custom name instead of uid

### DISCUSSION THREAD

**Discussion by tyrus-rex** [directus/directus#3364] [Feature Requests]
hello,

it should be possible to serve the file with the original asset name or a custom name. if the filename needs to be hashed, a prefix would be nice.

`GET /assets/{assetName}` or `GET /assets/{customName}` ?

for using in: 
```html
<div>
  <img src="http://www.mydomain.com/assets/my-seo-friendly-filename">
</div>
```

or does anyone have an idea how this can be realised with the existing tools? i could not find anything in the documentaiton.

**Comment by dimitrov-adrian** [comment]
What if the URL is `/assets/{uuid}/{filename_download}`

Anyway, I am thinking of practical usecase for this but not came up. There is an option ?download=1 that will tell the browser to download the file with set filename.

**Reply by tyrus-rex** [reply]
> What if the URL is `/assets/{uuid}/{filename_download}`

if that were possible the problem would be solved.

> Anyway, I am thinking of practical usecase for this but not came up. 

SEO Google Image best practices: The filename can give Google clues about the subject matter of the image. For example, my-new-black-kitten.jpg is better than IMG00023.JPG.
see more: https://developers.google.com/search/docs/advanced/guidelines/google-images

**Reply by rijkvanzanten** [reply]
I'm wondering what the actual SEO implications are of just the filename, compared to other more low-hanging fruit ones like the `alt` text, or content disposition header 🤔

**Reply by tyrus-rex** [reply]
since google does not openly communicate which search signal has which weight, this can only be conjectured. at least they mention it as one of the main features when seo optimizing images. 

- The people from **ahrefs.com** also explicitly point this out. https://ahrefs.com/blog/image-seo/
- **Yoast:** One of the most important seo wordpress plugin also states: "Image SEO starts with the file name." https://yoast.com/image-seo/#name

directus has a neat asset management, it would be great if there was the a feature like @dimitrov-adrian did point out or the option to use a custom id, a prefix or something else. seo agencies regularly ask for such adjustments on the customer side where we would love to use directus.

**Comment by connorwinston** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
